オブジェクト指向版ボーリングプラクティス

rira100000000 opened the pull request:
[ボウリングのスコア計算プログラムオブジェクト指向版](https://bootcamp.fjord.jp/practices/215)の提出物です。

rira100000000 committed:
計算機能を実装

ボーリングのスコア計算機能を実装しました。

rira100000000 committed:
コマンドライン引数の取得機能を追加

コマンドライン引数でボーリングのスコアを受け取れるようにしました。

rira100000000 committed:
不要なメソッドと変数の削除

不要なメソッドcalc_gameを削除しました
split_frameをcalc_gameにリネームしました
不要な変数one_shotを削除しました

rira100000000 committed:
frameクラスのリファクタリング

spare?メソッドをframeクラスに移動しました
strike?メソッドをframeクラスに追加しました

kaito0046 commented on 06.bowling_object/bowling.rb:
```diff
@@ -0,0 +1,16 @@
+end
+
+game = Game.new(option_perse[0])
+puts game.calc_game.sum
```
`game` がレシーバである時点で「 game に対してなにかする」というのは自明なので、`cale_game` というメソッド名は少し冗長かもしれません。`calc_何か` という形自体は良いと思うので、何を計算するのか、別の単語を当てはめて考えてみるといいかもです:eyes:（もちろん、もっとしっくりくる名前が思いつけばまったく別の形式でも良いです）

kaito0046 replied:
（メソッドを作るときは、メソッドの中身から書き始めるのではなく、呼び出し側から書くようにすると自然と上記のような点を意識しやすくなるのでおすすめです）

rira100000000 replied:
calc_gameをcalc_all_framesに修正します。
同じ観点でframeクラスのcalc_frameをcalc_scoresに修正します。

rira100000000 replied:
>呼び出し側から書くようにすると自然と上記のような点を意識しやすくなる

なるほどです！試してみます！

kaito0046 commented on 06.bowling_object/bowling.rb:
```diff
@@ -0,0 +1,16 @@
+require 'optparse'
+require_relative 'game'
+
+def option_perse
```
`parse` の typo でしょうか？

rira100000000 replied:
タイポでしたので修正します。

kaito0046 reviewed (commented):
ほぼよさそうです！
ロジックがシンプルで読みやすかったです〜

rira100000000 committed:
メソッド名の修正

frameクラスのcalc_frameをcalc_scoresに修正しました
gameクラスのcalc_gameをcalc_all_framesに修正しました
option_perseからoption_parseにタイポ修正しました

kaito0046 reviewed (approved):
LGTM！
